cleaned up rapid workflow and fire indicator exploration

ccmothes opened the pull request:
Breakdown of what's in here and what can be ignored:

**Rapid Workflow:**
- Updated the README with file structure and details. Use that as your reference for the new rapid workflow files.

**Code Review:**
- Most of the code review here are the new indicator functions in src/rapid. `scratch/explore_fire_indicators.Rmd` is the workflow for looking at the fire indicator visualizations I showed during our weekly meeting.
- Three new/updated functions for pulling park boundary information: `getparkSystem()` is just cleaned up and using the new file reference id. `getParkBoundary()` now just uses `getParkSystem()`. `getBoundaryArcGIS()` is a new function that can pull either park boundary or park tracts from the NPS ArcGIS REST Url. 

**Ignore:**
- Anything in rapid_app/ unless you want to see the shiny app structure. Note that the app_data/ folder is the data copied over from the results in data/rapid/all. It has to be on the repo for app publishing.
- Anything in the scratch/ folder except for explore_fire_indicators.Rmd (@kathryn-willi I moved over your original runoff function there as backup in case you/we ever want to reference it again).

KECognac commented on src/getBoundariesArcGIS.R:
this took FOREVER. Note @kathryn-willi, there is very little metadata associated with the tracts (such as helpful geographic names, etc) that would be helpful for the WSD :(. Function works great though -- just took multiple minutes to read in.

ccmothes replied:
Took ~1 minute for me, but that's expected with over 100K polygons.

KECognac commented on src/getParkBoundary.R:
great!

kathryn-willi commented on rapid_workflow.Rmd:
````diff
@@ -101,24 +171,192 @@ fire_exp <- calc_fire_exp(aoi = water_supplies, fp_data = delta_fireprob,
+# 2.77 mins for 1140 parks
+```
+
+Test function including moderate categories with high and very high:
````
suggest moving this exploratory section to the explore_fire_indicators.Rmd

kathryn-willi commented on src/rapid/helper_functions.R:
```diff
@@ -0,0 +1,354 @@
+
+
+
+# RUNOFF PROCESSING HELPER FUNCTIONS =====================================
```
Still not totally clear when we want to split up processing from the main `calc_` functions

ccmothes replied:
Me either....I was struggling with this decision but I think as we develop more indicators we will have a better idea of what would make the workflow cleaner and can adjust then!

kathryn-willi commented on scratch/explore_fire_indicators.Rmd:
For consistency, could you also move my loca2_mwbm_runoff_exploration.Rmd into the scratch folder? As we start doing these sorts of exploratory scripts for each component it might be nice to have them all located in one place :) (perhaps even in a scratch subfolder)

ccmothes commented on scratch/rapid_visualization.Rmd:
```diff
@@ -0,0 +1,81 @@
+#parks <- getParkSystem()
+#st_write(parks, paste0("data/rapid/all/park_boundaries_", Sys.Date(), ".gpkg"))
+parks <- st_read("data/rapid/all/park_boundaries_2025-08-14.gpkg")
+
```
This is just old scratch code I used to help put the shiny app vis together

ccmothes commented on scratch/vsicurl_ncfiles.R:
No, some personal scratch code I didn't want to part with yet lol.

ccmothes commented on rapid_workflow.Rmd:
```diff
@@ -12,47 +12,78 @@ source("setup.R")
+    source_longitude = as.numeric(source_longitude),
+    source_latitude = as.numeric(source_latitude)
+  ) %>%
+  filter(!state %in% c("AK", "AS", "HI", "MP", "PR", "VI")) %>%
```
Added st_transform(4326) to the left join because park_centroids is not in wgs84.
`left_join(
    park_centroids %>% 
      st_transform(4326) %>% 
      st_drop_geometry() %>% 
      dplyr::select(UNIT_CODE, x, y),
    by = c(park_unit = "UNIT_CODE")
  ) `

ccmothes commented:
I addressed comments in a few more commits. @kathryn-willi maybe most important for you is to just review my last commit "restructuring runoff indicator stuff" to see how I moved things around. Hopefully it makes sense/is helpful but we can totally redo some of this down the road.
